### PR TITLE
many CMakeLists.txt fixes for msvc

### PR DESCRIPTION
**Pull request by da2ce7**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -329,7 +373,7 @@ add_library(tunserver ${SOURCES_GROUP_TUNSERVER} ${SOURCES_GROUP_CRYPTO} ${SOURC
 
 
 if(MINGW)
-	set(MINGW_LIBRARIES ws2_32 iphlpapi)
+	set(MINGW_LIBRARIES "ws2_32 iphlpapi")
```
Comment (da2ce7) [inline CMakeLists.txt]: ah, ok will fix.

**Comment by da2ce7** [comment]
Updated from robet's comment.

File: CMakeLists.txt [hunk 2]
```diff
@@ -18,16 +18,55 @@ if(MINGW)
...
 	endif()
 endif()
 
-if (MINGW)
-	add_definitions(-D_WIN32_WINNT=0x0600)
+if(WIN32)
+  add_definitions(-DUNICODE)
+  add_definitions(-D_UNICODE)
+  add_definitions(-DWINVER=_WIN32_WINNT_WIN7)
+  add_definitions(-D_WIN32_WINNT=_WIN32_WINNT_WIN7)
```
Comment (robertoleksy) [inline CMakeLists.txt]: It is not working on MSVC with this line

5>c:\cygwin64\home\win2\galaxy42\src\c_tun_device.hpp(82): error C2039: 'stream_handle': is not a member of 'boost::asio::windows'
5>  C:\boost_1_61_0\boost/asio/windows/object_handle.hpp(28): note: see declaration of 'boost::asio::windows'
5>c:\cygwin64\home\win2\galaxy42\src\c_tun_device.hpp(82): error C2065: 'stream_handle': undeclared identifier
5>c:\cygwin64\home\win2\galaxy42\src\c_tun_device.hpp(82): error C2923: 'std::unique_ptr': 'stream_handle' is not a valid template type argument for parameter '_Ty'

Comment (da2ce7) [inline CMakeLists.txt]: fixed this build error. still other errors.

**Comment by da2ce7** [comment]
@robertoleksy @rfree2 I would love to get some review on this PR.

**Comment by rfree-d** [comment]
Reviewing it :)

**Comment by rfree-d** [comment]
Works on windows/MSCV; Though it seems that on linux it can not find lib boost. Looking into that now.

**Pull request closed without merging**